lib/client-assets.php

```diff
@@ -978,9 +972,17 @@ function gutenberg_register_vendor_scripts() {
 		'https://unpkg.com/element-closest@2.0.2/element-closest.js'
 	);
 	gutenberg_register_vendor_script(
-		'wp-polyfill-ecmascript',
+		'wp-polyfill',
 		'https://cdnjs.cloudflare.com/ajax/libs/babel-polyfill/7.0.0/polyfill' . $suffix . '.js'
 	);
+	// Ensure backwards compatibility after renaming to wp-polyfill.
```

Comment:
Is this something we want to do away with eventually? Should we make that clearer in the comment? How can we plan to follow-up?

Reply:
I'm honestly not sure best course of action here. Realistically, what is the likelihood that plugins are counting on this script handle? If we plan to eventually remove it, how long should we wait?

Reply:
We could follow the standard deprecation workflow:

https://wordpress.org/gutenberg/handbook/reference/deprecated/

We've done similar before for server-side features in either (a) enqueuing an inline script to run `wp.deprecated( '...' )` messaging or (b) [`_doing_it_wrong`](https://developer.wordpress.org/reference/functions/_doing_it_wrong/). The former seems easiest, since the second would be difficult in determining whether there are scripts which define a dependency on the script. 

Examples:

- https://github.com/WordPress/gutenberg/pull/10153/files#diff-620130744b6b73b822a3df609671d930L864
- https://github.com/WordPress/gutenberg/pull/8276/files#diff-620130744b6b73b822a3df609671d930L1214

Reply:
`wp-polyfill-promise` looks like a good example of prior art. What version should we set for removal?

Reply:
Two versions out, 4.5, which may or may not actually happen.